employees/tests/test_unit_report_viewset.py

```diff
@@ -148,3 +151,312 @@ def test_report_detail_view_should_delete_report_on_delete(self):
         response = ReportViewSet.as_view({'delete': 'destroy'})(request, pk=self.report.pk)
         self.assertEqual(response.status_code, 204)
         self.assertEqual(Report.objects.all().count(), 0)
+
+
+class ReportListTests(TestCase):
+    def setUp(self):
+        self.user = CustomUser(
+            email="testuser@codepoets.it",
+            password='newuserpasswd',
+            first_name='John',
+            last_name='Doe',
+            country='PL'
+        )
+        self.user.full_clean()
+        self.user.save()
+
+        self.project = Project(
+            name="Test Project",
+            start_date=datetime.datetime.now(),
+        )
+        self.project.full_clean()
+        self.project.save()
+
+        self.report = Report(
+            date=datetime.datetime.now().date(),
+            description='Some description',
+            author=self.user,
+            project=self.project,
+            work_hours=Decimal('8.00'),
+        )
+        self.report.full_clean()
+        self.report.save()
+        self.url = reverse('custom-report-list')
+
+    def test_custom_list_view_should_display_users_report_list_on_get(self):
+        request = APIRequestFactory().get(path=self.url)
+        request.user = self.user
+        response = ReportList.as_view()(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertContains(response, self.report.description)
+        dictionary = response.data['reports_dict']
+        reports = list(dictionary.values())[0]
```

Comment:
Like I said before, indexing is not supported with types returned by dictionary's functions.
In order to access an element with an index I have to use list conversion.

Would you like me to use actual keys instead? It would be either `reports = dictionary[datetime.datetime.now().date()]' or `reports = dictionary[self.report.date]`.

I just thought that it makes the code look very specific. Using numeric indexes with list conversion instead seemed to me like a way to make it easier for others to read through tests (because I'm numerically accessing the first value, rather than using "some key")